libs/framework/include/celix_framework.h

```diff
@@ -31,6 +31,25 @@
 extern "C" {
 #endif
 
+/**
+ * @file
+ * @brief The Celix Framework API.
+ *
+ * The Celix Framework API provides the functionality to create and destroy a Celix framework instance.
+ *
+ * The Celix framework instance provides the following functionality:
+ *  - Bundle management
+ *  - Service management
+ *  - Event management
+ *  - Logging
+ *  - Framework config properties
+ *  - Framework events
+ *  - Framework shutdown
+ *
+ * @note The Celix framework instance is thread safe.
```

Comment:
Yes, thread-safety should be guaranteed. 

When reading `celix_bundle_cache.c` I noticed that `readdir` is called without any protection, while `celix_framework_installBundle`, which changes cache structure, can be invoked from any threads. This is a sign of race-condition. We may need to re-consider the overall design.

Reply:
I added locking while accessing the bundle cache